src/tdastro/sources/periodic_variable_star.py

```diff
@@ -0,0 +1,243 @@
+        secondary_star_overlap_ratio = np.where(
+            primary_star_closer, overlap_area / (np.pi * self.secondary_radius**2), 0
+        )
+
+        primary_lum_density_eclipsed = primary_lum_density * (1 - primary_star_overlap_ratio)
+        secondary_lum_density_eclipsed = secondary_lum_density * (1 - secondary_star_overlap_ratio)
+
+        # Just in case, we clip negative values to zero
+        total_lum_density = np.where(
+            primary_lum_density_eclipsed >= 0, primary_lum_density_eclipsed, 0
+        ) + np.where(secondary_lum_density_eclipsed >= 0, secondary_lum_density_eclipsed, 0)
+
+        return total_lum_density
+
+    @staticmethod
+    def _norm_star_center_distance(phase_fraction, inclination_degree):
+        """Calculate the distance between the centers of the stars on the plane of the sky.
+
+        Parameters
+        ----------
+        phase_fraction : `np.ndarray`
+            The phase of the orbit, in the range [0, 1].
+        inclination_degree : `np.ndarray`
+            The inclination of the orbit, in degrees.
+
+        Returns
+        -------
+        distance : `no.ndarray`
+            The distance between the centers of the stars on the plane of the sky,
+            normalized by the major semiaxis.
+        """
+        phase_radians = 2 * np.pi * phase_fraction
+        inclination_radians = np.radians(inclination_degree)
+        return np.hypot(np.cos(inclination_radians) * np.cos(phase_radians), np.sin(phase_radians))
+
+    # Math description can be found here:
+    # https://math.stackexchange.com/a/3543551/1348501
+    @staticmethod
+    def _circle_overlap_area(d, r1, r2):
+        """Calculate the area of overlap between two circles.
+
+        Parameters
+        ----------
+        d : `np.ndarray`
+            The distance between the centers of the circles.
+        r1 : `float`
+            The radius of the first circle.
+        r2 : `float`
+            The radius of the second circle.
+
+        Returns
+        -------
+        overlap_area : `np.ndarray`
+            The area of overlap between the two circles.
+        """
+        r1_sq = r1**2
+        r2_sq = r2**2
+        d_sq = d**2
+
+        # Avoid invalid values for arccos and sqrt, we will handle the invalid values later
+        with np.errstate(invalid="ignore"):
+            term1 = r1_sq * np.arccos((d_sq + r1_sq - r2_sq) / (2 * d * r1))
+            term2 = r2_sq * np.arccos((d_sq + r2_sq - r1_sq) / (2 * d * r2))
+            term3 = 0.5 * np.sqrt((-d + r1 + r2) * (d + r1 - r2) * (d - r1 + r2) * (d + r1 + r2))
+
+        area = term1 + term2 - term3
+        # Circles do not overlap
+        area = np.where(d >= r1 + r2, 0, area)
+        # Circles fully overlap
+        minimum_radius = np.minimum(r1, r2)
+        area = np.where(d <= minimum_radius, np.pi * minimum_radius**2, area)
+
+        return area
+
+
+# We should move this function to a more appropriate location
```

Comment:
Maybe we create a directory `src/tdastro/astro_utils` for this and other functions?

Reply:
I’d leave it here for now; currently, it is a toy model and an example implementation.